netsim/ansible/templates/stp/cumulus.j2

```diff
@@ -36,6 +36,9 @@ iface {{ ifdata.ifname }}
 # Newer versions 5.x support mstpctl-port-vlan-priority <vlan id>={{ ifdata.stp.port_priority }}
 #
 {%  endif %}
+{%  set port_type = ifdata.stp.port_type|default("normal") %}
```

Comment:
Wouldn't it be better to start with `if ifdata.stp.port_type is defined`? Otherwise, if you insist on using stp.port_type without checking whether it's defined, it might be better to set the default value in the STP module instead of having `default(normal)` filter in every single template.

Also, this effectively overrides device default value even in case where the user does not care (or does not want to change the value).

Applies to all templates.

Reply:
added ```if defined```